t/body_utf8.t

```diff
@@ -7,75 +7,63 @@ use Email::MIME;
-    [ $e, $e, "big5-hkscs", "bytes in unknown encoding instead of big5-hkscs" ],
+	[ $ascii_string, $ascii_string, "UTF-8" ],    #
+	[ $perlstring,   $perlstring,   "UTF-8" ],
+	[ $encoded_utf8, $perlstring,  "UTF-8",      "bytes in utf8 encoding" ],
+	[ $encoded_jis,  $perlstring,  "shiftjis",   "bytes in shiftjis encoding" ],
+	[ $encoded_jis,  $encoded_jis, "big5-hkscs", "bytes in unknown encoding instead of big5-hkscs" ],
   )
 {
-    my ( $str, $expected, $encoding, $status_expected ) = $_->@*;
-    my ( $res, $status ) = Email::MIME::CreateHTML::_normalize_to_perl_string( $str, $encoding );
-    is $res,    $expected,        "output";
-    is $status, $status_expected, "status";
+	my ( $str, $expected, $encoding, $status_expected ) = $_->@*;
+	my ( $res, $status ) = Email::MIME::CreateHTML::_normalize_to_perl_string( $str, $encoding );
+	is $res,    $expected,        "output";
+	is $status, $status_expected, "status";
 }
 
 my @core = ( header => [ From => 'unittest_a@example.co.uk', To => 'mail@mail', Subject => '.' ] );
 my @base = ( @core, body => );
 
-my ( undef, $warn ) = capture { Email::MIME->create_html( @base, $d ) };
+my ( undef, $warn ) = capture { Email::MIME->create_html( @base, $encoded_utf8 ) };
 like $warn, qr/created email may be corrupt, body was not a decoded perl unicode string, but: bytes in utf8 encoding/, "warns with encoded body";
 
-( undef, $warn ) = capture { Email::MIME->create_html( @base, $c ) };
+( undef, $warn ) = capture { Email::MIME->create_html( @base, $perlstring ) };
 is $warn, "", "decoded perl unicode string causes no warnings";
 
-( undef, $warn, my $res ) = capture { Email::MIME::CreateHTML->create( @core, body_str => $c ) };
-is $res->body_str, $c, "body didn't get corrupted";
+( undef, $warn, my $res ) = capture { Email::MIME::CreateHTML->create( @core, body_str => $perlstring ) };
+is $res->body_str, $perlstring, "body didn't get corrupted";
 is $warn, "", "decoded perl unicode string causes no warnings";
 
-( undef, $warn, $res ) = capture { Email::MIME::CreateHTML->create( @core, body => $d ) };
-is $res->body_str, $c, "body didn't get corrupted";
+( undef, $warn, $res ) = capture { Email::MIME::CreateHTML->create( @core, body => $encoded_utf8 ) };
+is $res->body_str, $perlstring, "body didn't get corrupted";
 is $warn, "", "decoded perl unicode string causes no warnings";
 
-( undef, $warn, $res ) = capture { Email::MIME::CreateHTML->create( @core, body => $e, body_attributes => { charset => "shiftjis" } ) };
-is $res->body_str, $c, "body didn't get corrupted";
+( undef, $warn, $res ) = capture { Email::MIME::CreateHTML->create( @core, body => $encoded_jis, body_attributes => { charset => "shiftjis" } ) };
+is $res->body_str, $perlstring, "body didn't get corrupted";
 is $warn, "", "decoded perl unicode string causes no warnings";
 
 subtest text_body => sub {
-	my @methods= (
-		[ "via old create API",
-		  sub { Email::MIME->create_html( @core,

```

Comment:
One detail got lost here; I was calling Email::MIME->create_html for this one

Reply:
The differences between those two calls were: A bug due to the $class variable, which is eradicated by 59c2f7a, and whether the encoding guessing is being done. With the inputs given that latter part isn't important either.